internal/database/repository/node_int_test.go

```diff
@@ -8,29 +8,52 @@ import (
 	"github.com/G-Research/yunikorn-core/pkg/webservice/dao"
 	"github.com/oklog/ulid/v2"
 	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 
 	"github.com/G-Research/unicorn-history-server/internal/model"
 
 	"github.com/G-Research/unicorn-history-server/internal/util"
 	"github.com/G-Research/unicorn-history-server/test/database"
 )
 
-func TestGetNodesPerPartition_Integration(t *testing.T) {
-	if testing.Short() {
-		t.Skip("skipping integration test in short mode")
-	}
+type NodeTestSuite struct {
+	suite.Suite
+	tp   *database.TestPostgresContainer
+	repo *PostgresRepository
+}
 
+func (ns *NodeTestSuite) SetupSuite() {
```

Comment:
Do we want to run many test suites? Isn't it better to have a single db instance and maybe, if we really need to do it, create different databases?

Reply:
Thanks. I thought creating multiple db containers would help to run the tests in parallel. But the idea of creating multiple db in a single container sounds good.

Reply:
updated with separate DB instead of separate container